### PR TITLE
Modifications for metaproject with stateless config manager

### PR DESCRIPTION
**Pull request by rsgoncalves**

### CODE REVIEW

File: src/main/java/org/protege/editor/owl/server/base/ProtegeServer.java [hunk 1]
```diff
@@ -95,7 +93,7 @@ public void deleteUser(AuthToken token, UserId userId) throws AuthorizationExcep
             try {
                 User user = configuration.getUser(userId);
                 logger.info(printLog(token.getUser(), "Remove user", user.toString()));
-                manager.removeUser(user);
+                configuration = new ConfigurationBuilder(configuration).removeUser(user).createServerConfiguration();
```
Comment (bdionne) [inline src/main/java/org/protege/editor/owl/server/base/ProtegeServer.java]: this call no longer throws `UnknownUserIdException`

File: src/main/java/org/protege/editor/owl/server/base/ProtegeServer.java [hunk 2]
```diff
@@ -184,7 +172,7 @@ public void deleteProject(AuthToken token, ProjectId projectId, boolean includeF
             try {
                 Project project = configuration.getProject(projectId);
                 logger.info(printLog(token.getUser(), "Remove project", project.toString()));
-                manager.removeProject(project);
+                configuration = new ConfigurationBuilder(configuration).removeProject(project).createServerConfiguration();
```
Comment (bdionne) [inline src/main/java/org/protege/editor/owl/server/base/ProtegeServer.java]: this also no longer throws `UnknownProjectIdException`

**Comment by bdionne** [comment]
all smoke tests run - LGTM

File: src/main/java/org/protege/editor/owl/server/base/ProtegeServer.java (hunk 1 above)
Comment (rsgoncalves) [inline src/main/java/org/protege/editor/owl/server/base/ProtegeServer.java]: The removeXX() no longer throw these exception, but they are still thrown by the getUser(id) and getProject(id) methods.

**Pull request merged**